docs/http-request.md

```diff
@@ -0,0 +1,41 @@
+# The life of a `code.org` HTTP Request
+
+## DNS resolution
+
+`code.org` => (DNS managed by route 53) => `54.192.37.35` (...etc...)
+
+(`dig code.org` => `54.192.37.35`, `whois 54.192.37.35` <sup>[output](https://gist.github.com/wjordan/1f2c60e3fec28dbef5c774f08fb3c675)</sup> shows it's an Amazon-owned IP, `AMAZO-CF2` == CloudFront.)
+
+## CDN / load-balancer
+
+[CloudFront](https://aws.amazon.com/cloudfront/) => [Elastic Load Balancing](https://aws.amazon.com/elasticloadbalancing/) [ELB]
+
+ELB => => multiple Frontend servers ([Elastic Compute Cloud](https://aws.amazon.com/ec2/) [EC2] with [Auto Scaling](https://aws.amazon.com/autoscaling/))
+
+## Frontend Server
+[Varnish Cache](https://varnish-cache.org/) (port `80`) => port `8080` (dashboard) or port `8081` (pegasus) (depending on `Host` header: `code.org` => `pegasus`, `studio.code.org` => `dashboard`)
+
+[NGiNX](https://www.nginx.com/) (port `808{0,1}`) => Unix socket/s (`/run/unicorn/{dashboard,pegasus}.sock`)
+
+[Unicorn](https://bogomips.org/unicorn/) (UNIX socket/s) => Ruby application Rack server/s
+
+Rack server definition files:
+- dashboard: [`dashboard/config.ru`](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/config.ru)
+- pegasus: [`pegasus/config.ru`](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/config.ru)
+
+### Pegasus application (`/pegasus`)
+
+Web-application based on [`sinatra`](http://www.sinatrarb.com/) routing framework, with custom template and view-rendering layer.
+- [`pegasus/router.rb`](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/router.rb): - `class Documents < Sinatra::Base`
```

Comment:
Might want to add, additional routes are in [pegasus/routes](https://github.com/code-dot-org/code-dot-org/tree/staging/pegasus/routes) (which is included in router.rb). This is where new ones should be added.